### PR TITLE
Enhancement/251 migrate from bower to npm

### LINKED ISSUE #251: Migrate bower + npm to only npm for dependency management
No reason to have both of these dependency managers around, especially given #245 we stand to gain much from using npm.

This task is broken up into two parts:
- [ ] Transfer bower dependencies to package.json
- [ ] Ensure necessary dependencies are injected into dependency partials (gulp task)

### PR DESCRIPTION
**Pull request by AllenBW**
☠️  bower, long live bower 👑 

Re-configures process for adding dependencies, need to add documentation on this, was thinking of starting a `guides` folder in where I can explain the process (long story short npm install, add fep location in the stylesheet.html or javascript.html, update test config dependencies to reflect)

Renames a number of gulp tasks to better align with actual function
Removes a number of unnecessary gulp tasks relating to building the test environment

End of the day, not much changed here! Though I would love some 👀 from the following :godmode: :
@chriskacerguis @himdel @jeff-phillips-18 @dtaylor113 

closes #251

### CODE REVIEW

**Comment by AllenBW** [comment]
@miq-bot add_label enhancement

File: client/app/components/components.module.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
     'app.core',
     'ui.bootstrap',
     'patternfly',
-    'ngSVGAttributes',
+    'svgBaseFix',
```
Comment (dtaylor113) [inline client/app/components/components.module.js]: Cloned this branch, new svgBaseFix lib is working.  LGTM

File: .travis.yml [hunk 2]
```diff
@@ -6,10 +6,9 @@ cache:
   directories:
   - node_modules
 before_install:
-- npm install -g yarn gulp bower
+- npm install -g yarn gulp
 install:
-- yarn install
-- bower install -F
+- yarn install --force
```
Comment (chriskacerguis) [inline .travis.yml]: Why --force?  Using --force could do very bad things

Comment (AllenBW) [inline .travis.yml]: Thats a good question, travis 💩's if its absent, I would say that force isn't so bad, just fetches deps fresh! https://yarnpkg.com/en/docs/cli/install#toc-yarn-install-force

**Comment by chriskacerguis** [comment]
👍 for me, waiting for @himdel and @martinpovolny to take a look.

File: gulp/config.js [hunk 3]
```diff
@@ -83,7 +69,57 @@ module.exports = (function() {
     var options = {
       files: [].concat(
         './node_modules/phantomjs-polyfill/bind-polyfill.js',
-        wiredep({devDependencies: true}).js,
+        './node_modules/jquery/dist/jquery.js',
```
Comment (himdel) [inline gulp/config.js]: This is problematic ;). Is there any way to make `wiredep` work? Is it just that wiredep can't read package.json or pick the right libraries, or is it also that those don't have the necessary info on what to include?

People will notice that adding the lib is not enough and add it to javascripts, but.. these test deps will get out of sync soon, I'm afraid..

**Comment by himdel** [comment]
When running `gulp build`, I'm getting

```
[05:50:39] gulp-inject 1 files into index.html.
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/bootstrap-select/dist/css/bootstrap-select.css
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/bootstrap-switch/dist/css/bootstrap3/bootstrap-switch.css
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/bootstrap-touchspin/src/jquery.bootstrap-touchspin.css
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/c3/c3.css
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/google-code-prettify/bin/prettify.min.css
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/google-code-prettify/bin/prettify.min.js
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/c3/c3.js
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/bootstrap-datepicker/dist/js/bootstrap-datepicker.min.js
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/bootstrap-select/dist/js/bootstrap-select.js
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/bootstrap-switch/dist/js/bootstrap-switch.js
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/bootstrap-touchspin/src/jquery.bootstrap-touchspin.js
[05:50:39] Plumber found unhandled error:
 Error: Error: File not found with singular glob: /home/himdel/manageiq-ui-service/node_modules/patternfly-bootstrap-treeview/dist/bootstrap-treeview.min.js
```

And looks like it's right, those modules' directories are empty, except for a `node_modules/jquery`..

Either way, `gulp build` fails .. is it just me?

**Comment by himdel** [comment]
This looks good, except for that manual test dependency list.. Any chance we could include `javascripts.html` in the spec html as well instead of that?

File: server/utils/serviceApp.js [hunk 4]
```diff
@@ -9,6 +9,6 @@ var router = express.Router();
 
 router.use(express.static('./client'));
 router.use(express.static('./.tmp'));
-router.use(express.static('./bower_components'));
+router.use(express.static('./node_modules'));
```
Comment (himdel) [inline server/utils/serviceApp.js]: Console templates ;-)

File: gulp/config.js (hunk 3 above)
Comment (AllenBW) [inline gulp/config.js]: wiredep is for bower dependencies only ever just bower dependencies, i TOTALLY feel what you are feeling, wait what, more than one place to update deps?  I kinda ran outta time and ideas on how to getting those in here, open to suggestions!  

ultimately i was going to write up some documentation on how to correctly add dependencies and keep in mind, this is only until we move to angular 2!  importing modules will change a great deal of what we presently do...

**Comment by AllenBW** [comment]
@himdel sooooooo we can't require the file because `javascripts.html` is a bunch of scripts rather than a bunch of file locations (which is required in the tests)

silver lining, if ever the list starts to deteriorate, tests will fail, chaos will FOLLOW

File: server/utils/serviceApp.js (hunk 4 above)
Comment (AllenBW) [inline server/utils/serviceApp.js]: oh yeah soooo those served from a folder renamed `vendor` (also pushed to that folder)
but yeah that changed  whos da person to beep for that change?

Comment (himdel) [inline server/utils/serviceApp.js]: Uh, I guess that would be me :D 

Ok, I'm seeing they're now copied to `public/ui/service/vendor/{no-vnc,spice-html5-bower}`.. The tricky part here is that we need to find them on the same url both in devel mode (served by node) and in production mode (served by apache) .. but that can be acomplished by redirecting `/ui/service/vendor` to `/node_modules` in node, and assuming the `/ui/service/vendor/...` URL, I think. 

And then, the only change needed outside manageiq-ui-service would be https://github.com/ManageIQ/manageiq/blob/3921e87915b5a69937b9d4a70bb24ab71b97c165/gems/pending/util/miq_apache/miq_apache.rb#L209 .. that `bower_components` was there for exactly that reason (and only that reason, so you should be able to safely change that to `vendor`)

Comment (AllenBW) [inline server/utils/serviceApp.js]: @himdel acceptable for me to submit a second pr making the above change? or is this a thing fun enough to rope @simaishi into?  👐 🙌

Comment (himdel) [inline server/utils/serviceApp.js]: I'd say feel free to do it, the more people understand these things the better :-) but do tag us there.. :-)

**Comment by Fryguy** [comment]
cc @simaishi for build concerns

**Pull request merged**